static/app/components/tooltip.tsx

```diff
@@ -116,215 +111,262 @@ function computeOriginFromArrow(
+ */
+async function getAcceptanceTestTooltipStore() {
+  const {tooltipStore} = await import('sentry/stores/tooltipStore');
+  return tooltipStore;
+}
+
+type UseTooltipInAcceptanceTestsOtpions = {
+  ignore: boolean;
+  setOpen: (open: boolean) => void;
+  setUseGlobalPortal: (useGlobalPortal: boolean) => void;
+  tooltipId: string;
+};
+
+/**
+ * When running the acceptance tests, we expose functionality to open / close
+ * all of the tooltips. This hook handles registration of tooltips in the store,
+ * with special care to unregister tooltips if they become ignored.
+ */
+function useTooltipInAcceptanceTests({
+  tooltipId,
+  setOpen,
+  setUseGlobalPortal,
+  ignore,
+}: UseTooltipInAcceptanceTestsOtpions) {
+  if (!IS_ACCEPTANCE_TEST) {
+    return;
   }
 
-  tooltipId: string = domId('tooltip-');
-  delayTimeout: number | null = null;
-  delayHideTimeout: number | null = null;
-  triggerEl: Element | null = null;
-
-  getPortal = memoize((usesGlobalPortal): HTMLElement => {
-    if (usesGlobalPortal) {
-      let portal = document.getElementById('tooltip-portal');
-      if (!portal) {
-        portal = document.createElement('div');
-        portal.setAttribute('id', 'tooltip-portal');
-        document.body.appendChild(portal);
-      }
-      return portal;
-    }
-    const portal = document.createElement('div');
-    document.body.appendChild(portal);
-    return portal;
-  });
+  async function registerTooltip() {
+    const tooltipStore = await getAcceptanceTestTooltipStore();
+    tooltipStore?.addTooltip(tooltipId, {setOpen, setUseGlobalPortal});
+  }
 
-  setOpen = () => {
-    this.setState({isOpen: true});
-  };
+  async function unregisterTooltip() {
+    const tooltipStore = await getAcceptanceTestTooltipStore();
+    tooltipStore?.removeTooltip(tooltipId);
+  }
 
-  setClose = () => {
-    this.setState({isOpen: false});
-  };
+  // Register and unregister tooltips from the store
+  useEffect(() => {
+    if (!ignore) {
+      registerTooltip();
+    }
+    return () => void unregisterTooltip();
+  }, [ignore]);
+}
 
-  handleOpen = () => {
-    const {delay, showOnlyOnOverflow} = this.props;
+function Tooltip({
+  children,
+  className,
+  delay,
+  forceShow,
+  isHoverable,
+  popperStyle,
+  showOnlyOnOverflow,
+  skipWrapper,
+  title,
+  disabled = false,
+  disableForVisualTest = false,
+  position = 'top',
+  containerDisplayMode = 'inline-block',
+}: Props) {
+  const [isOpen, setOpen] = useState(false);
+
+  // XXX: We use this when running acceptance tests with 'open all tooltips'
+  const [useGlobalPortal, setUseGlobalPortal] = useState(true);
+
+  // Tooltip ID is stable accross renders
+  const tooltipId = useMemo(() => domId('tooltip-'), []);
+
+  // Delayed open and close time handles
+  const delayOpenTimeout = useRef<number | null>(null);
+  const delayHideTimeout = useRef<number | null>(null);
+
+  // Tracks the triggering element
+  const triggerEl = useRef<HTMLElement | null>(null);
```

Comment:
```suggestion
  const triggerElementRef = useRef<HTMLElement | null>(null);
```

Reply:
Gonna go with `TriggerRef` to keep some longer lines from wrapping